.github/workflows/analyzer.yml

```diff
@@ -0,0 +1,32 @@
+name: Analyze Code
+
+on:
+  push:
+    branches: dev
+    paths: 
+      - "**.dart"
+      - ".github/workflows/analyze.yml"
```

Comment:
I don't think on push is necessary, the code is already pushed

And this would duplicate runs on PRs from dev to main, which already covers the use of this

Reply:
There are some cases where we would wanna push directly to the repository rather make a pull request, could be situations like—critical bug fixes, strings changes, miscellaneous stuff.

We would wanna have the analyser running at all time so that we can check if one of our commit can causes build to fail.